### PR TITLE
Update publish command

### PR DESCRIPTION
**Pull request by jonlouie**
#### Description of change
* Update `dotnet publish` command (starting with projects targeting .NET 5, the publish command behaves differently)
  * Remove explicit target of netcoreapp3.1 (will target the runtime in the .csproj file)
  * Set param `-p:IncludeAllContentForSelfExtract=true` to maintain similar output structure as .netcoreapp3.1
  * Set param `-p:IncludeSymbolsInSingleFile=false` as it is no longer supported beyond netcoreapp3.1
* ~~Replace calls to `Assembly.GetExecutingAssembly().Location` (starting with .NET 6, these calls return an empty string if called from within a single file app)~~ These changes were reverted as they are no longer needed with the latest publish command
* Replace deprecated SHA256 api

#### PR reviewer notes
* The new publish command was tested by running the packaged .exe through our CLI unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Eruanion — APPROVED** [review]

**Review by xueningl-az — DISMISSED** [review]
LGTM

File: tests/PortingAssistant.Client.UnitTests/PortingAssistantNugetHandlerTest.cs [hunk 1]
```diff
@@ -306,7 +306,7 @@ public void Setup()
             _fileSystem.Setup(fileSystem => fileSystem.FileExists(It.IsAny<string>())).Returns((string file) =>
             {
                 string solutionId;
-                using (var sha = new SHA256Managed())
+                using (var sha = SHA256.Create())
```
Comment (saikiranakula-amzn) [inline tests/PortingAssistant.Client.UnitTests/PortingAssistantNugetHandlerTest.cs:309]: did you check remaining parts of the code and also change there, seems like SHA256Managed is obselete

File: src/PortingAssistant.Client/Program.cs [hunk 2]
```diff
@@ -140,6 +146,17 @@ static void Main(string[] args)
             }
         }
 
+        private static string GetAppEntryPoint()
+        {
+            var executingDirectory = AppContext.BaseDirectory;
+            var assemblyFile = Path.Combine(executingDirectory, DllName);
+            var executableFile = Path.Combine(executingDirectory, ExeName);
+
+            var entryPointFile = Directory.EnumerateFiles(executingDirectory, $"{ProjectName}.*")
```
Comment (saikiranakula-amzn) [inline src/PortingAssistant.Client/Program.cs:155]: have a null check here just incase "executingDirectory" doesn't contain files with the name, and also use FirstorDefault

File: tests/PortingAssistant.Client.UnitTests/PortingAssistantNugetHandlerTest.cs (hunk 1 above)
Comment (jonlouie) [reply tests/PortingAssistant.Client.UnitTests/PortingAssistantNugetHandlerTest.cs:309]: Yes, I checked the rest of the code to replace the obsolete api. Only found 1 reference in src and 2 in the test directory

File: src/PortingAssistant.Client/PortingAssistant.Client.CLI.csproj [hunk 3]
```diff
@@ -3,7 +3,6 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>net6.0</TargetFramework>
-	<IncludeSymbolsInSingleFile>true</IncludeSymbolsInSingleFile>
```
[the remark this answers is not in the document]
Comment (jonlouie) [reply src/PortingAssistant.Client/PortingAssistant.Client.CLI.csproj:6]: During regular compile/build, we do not create the SingleFile package.

`IncludeSymbolsInSingleFile` is not supported starting with .NET 5 so either way, this needs to be removed.

**Review by Eruanion — APPROVED** [review]

**Review by saikiranakula-amzn — APPROVED** [review]

**Review by cslong — APPROVED** [review]

**Review by huawenm — APPROVED** [review]

File: src/PortingAssistant.Client/Program.cs (hunk 2 above)
Comment (Mingxue008) [reply src/PortingAssistant.Client/Program.cs:172]: This part is confusing a bit, this function is to return either a dll file or an exe file. 
What happens if both exists?
How about neither exists, the default value is an empty string then will it crash the caller of this function?

File: src/PortingAssistant.Client/Program.cs [hunk 4]
```diff
@@ -71,16 +93,17 @@ static void Main(string[] args)
...
-                            new AnalyzerSettings
-                            {
-                                IgnoreProjects = cli.IgnoreProjects,
-                                TargetFramework = cli.Target
-                            } : new AnalyzerSettings
-                            {
-                                IgnoreProjects = new List<string>(),
-                                TargetFramework = cli.Target
-                            };
+                    var solutionSettings = cli.IgnoreProjects != null && cli.IgnoreProjects.Count != 0
```
Comment (Mingxue008) [inline src/PortingAssistant.Client/Program.cs:96]: Thanks for the new formatting, looks much better.
Nitpick: My coding habit would be adding parentheses around this (A && B) so it's more straightforward and readable.

**Review by cslong — APPROVED** [review]

**Review by Beau-Gosse-dev — APPROVED** [review]

**Pull request merged**